### ISSUE TITLE
jx upgrade cli after a manual install (not via brew) fails

### ISSUE THREAD

**Issue by agentgonzo** [jenkins-x/jx#3023]
### Summary 
If you have previously installed jx via brew, then uninstalled it and the manually installed it via the download, a `jx upgrade cli` fails.

### Steps to reproduce the behavior
`brew install jx`
`brew uninstall jx`
Download and install jx manually
`jx upgrade cli`

### Expected behavior
the manually installed version of jx should be upgraded

### Actual behavior
it tries to install via brew, which fails
![image](https://user-images.githubusercontent.com/1652511/52411651-72dc8a00-2ad4-11e9-97da-3f7233d8a696.png)

### Jx version

The output of `jx version` is:

```
COPY OUTPUT HERE
![image](https://user-images.githubusercontent.com/1652511/52411660-76701100-2ad4-11e9-91b8-652dc46cd0f5.png)
```

### Jenkins type

- [ ] Classic Jenkins
- [ ] Serverless Jenkins

### Operating system / Environment
mac

**Comment by msigler-cb** [comment]
First, remove jx using `brew brew uninstall jx`. Do `brew update` and `brew list` and not see jx. Also suggest `brew cleanup`. Then install per the guide: https://go.cloudbees.com/docs/cloudbees-core/kubernetes-cd-install-guide/install-base/ . This would include `curl -L https://github.com/jenkins-x/jx/releases/download/v1.3.620/jx-linux-amd64.tar.gz | tar xzv` and `sudo mv jx /usr/local/bin`. This should remove any preexisting association with brew, and install current version of jx without using brew. Any future upgrades should be using `jx upgrade cli --no-brew` which should prevent unintended upgrades with the rest of `brew update && brew upgrade`.  

This MacOS specific bug may also apply to any other client package manager.

**Comment by msigler-cb** [comment]
I suspect the next time I do `jx version` and there is an upgrade available, that i will be prompted, and with `Y` i will get the brew error.

**Comment by ccojocar** [comment]
Please could you try to run `brew upgrade jx` outside of jx? I suspect you will get something like this:

```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/jx
Target /usr/local/bin/jx
already exists. You may want to remove it:
  rm '/usr/local/bin/jx'

To force the link and overwrite all conflicting files:
  brew link --overwrite jx

To list all files that would be deleted:
  brew link --overwrite --dry-run jx
```

**Comment by jstrachan** [comment]
maybe rather than assuming brew was used to install the binary we should look at the location of the binary and use that to determine if we should even bother with brew? If the binary isn't somewhere brew would put it - we just ignore brew?

**Comment by msigler-cb** [comment]
I happen to know that it was installed using brew, but I don't think it was `brew uninstall jx` prior to `jx upgrade cli` so brew still has it in it's manifest

brew also installs jx in /usr/local/bin path, so i don't think you can determine definitively by path

`brew list jx` will "Error: No such keg: /usr/local/Cellar/jx"

`brew upgrade jx` will "Error: jenkins-x/jx/jx not installed"

The intention is to remove jx from brew and not use brew for upgrades as this might upgrade during a `brew update && brew upgrade` enmasse. This way we can control upgrade of jx discrete from other packages under brew on the client.

**Comment by agentgonzo** [comment]
>The intention is to remove jx from brew and not use brew for upgrades as this might upgrade during a `brew update && brew upgrade` enmasse. 

If I was to do a `brew update && brew upgrade`, that is exactly what I would _want_ to if I had installed `jx` via brew happen.

>This way we can control upgrade of jx discrete from other packages under brew on the client.

As an end-user, I would _not want_ this.

**Comment by agentgonzo** [comment]
FYI, I wasn't the originator of this and don't own a mac, so can't help diagnose issues :-|

**Comment by agentgonzo** [comment]
>I happen to know that it was installed using brew, but I don't think it was `brew uninstall jx` prior to `jx upgrade cli` so brew still has it in it's manifest

Can we get some more clarification on _exactly_ what the user did to get themselves into this state? I am starting to think that it's a situation where the user has done something unexpected/unsupported and therefore _potentially_ outside the bounds of what we can reasonably code for

**Comment by msigler-cb** [comment]
Didn't create a new bug as these are related...

NAME               VERSION
jx                 1.3.853

My CLI was not installed by brew.

Help text "--no-brew=false: Disables the use of brew on macOS to install or upgrade command line dependencies"

$ jx upgrade cli --no-brew=false
Error: Command failed  brew upgrade jx
error: exit status 1

$ jx upgrade cli --no-brew
Downloading https://github.com/jenkins-x/jx/releases/download/v1.3.967/jx-darwin-amd64.tar.gz to /usr/local/bin/jx.tmp...
Downloaded /usr/local/bin/jx.tmp
Jenkins X client has been installed into /usr/local/bin/jx

$ jx version
NAME               VERSION
jx                 1.3.967

I think this may be simply a case for updating the help text, as i think no-brew=true would disable.

**Comment by garethjevans** [comment]
This should be resolved in 2.0.277
